### PR TITLE
add overview of changes in EasyBuild v4.0

### PR DESCRIPTION
**Pull request by boegel**
Preview:

* https://boegel-eb.readthedocs.io/en/eb4_changes_overview/EasyBuild4-overview-of-changes.html
  * https://boegel-eb.readthedocs.io/en/eb4_changes_overview/Python-2-3-compatibility.html
  * https://boegel-eb.readthedocs.io/en/eb4_changes_overview/System_toolchain.html
  * https://boegel-eb.readthedocs.io/en/eb4_changes_overview/EasyBuild4-overview-relocated-functions-constants.html

### CODE REVIEW

File: docs/Manipulating_dependencies.rst [hunk 1]
```diff
@@ -200,7 +200,7 @@ So, for the zlib v1.2.8 dependency included in the example above, the following
 * if (only) ``--minimal-toolchains`` is enabled, EasyBuild will search for an easyconfig file for
   zlib v1.2.8 using the ``GCC/4.9.3`` toolchain; if no such easyconfig file is found, it will continue searching
   using the ``gompi/2015b`` toolchain, and finally the ``foss/2015b`` toolchain
-* if ``--add-dummy-to-minimal-toolchains`` is also enabled, EasyBuild will try locating an easyconfig file for
+* if ``--add-system-to-minimal-toolchains`` is also enabled, EasyBuild will try locating an easyconfig file for
   zlib v1.2.8 that uses the ``dummy`` toolchain prior to consider the ``GCC/4.9.3`` toolchain
```
Comment (akesandgren) [inline docs/Manipulating_dependencies.rst]: Shouldn't that be "system" toolchain?

File: docs/Python-2-3-compatibility.rst [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+More specifically, the following Python versions are currently supported:
+
+* Python 2.6.x
+* Python 2.7.x
+* Python 3.5.x
+* Python 3.6.x
+* Python 3.7.x
+
+To facilitate this, the ``easybuild.tools.py2vs3`` package was introduced in EasyBuild v4.0.
+When importing a function from this package, you will automagically get a version of the function that
```
Comment (akesandgren) [inline docs/Python-2-3-compatibility.rst]: hmmm "automatically" is probably better

File: docs/System_toolchain.rst [hunk 3]
```diff
@@ -0,0 +1,148 @@
+.. _system_toolchain:
+
+``system`` toolchain
+====================
+
+In EasyBuild v4.0, the infamous ``dummy`` toolchain was deprecated, and replaced with the ``system`` toolchain.
```
Comment (akesandgren) [inline docs/System_toolchain.rst]: "was deprecated" -> "has been deprecated" or is it "have been"?

File: docs/System_toolchain.rst [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+This page provides more information about this change, and how to deal with the transition.
+
+.. note:: To easy the transition from ``dummy`` to ``system``, EasyBuild v3.9.4 already supports using the ``system``
```
Comment (akesandgren) [inline docs/System_toolchain.rst]: "To easy" -> "To ease"

File: docs/System_toolchain.rst [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+Use case for the ``system`` toolchain
+-------------------------------------
+
+The :ref:`system_toolchain` is intended for the same use case as the ``dummy`` toolchain was: it should (only) be used
+for software installations where the compiler & libraries provided by the operating system are assumed to
+be sufficient.
+
+This includes installing binary software (for which no compiler is used at all), installations that boil down to
+only unpacking a source tarball, and building and installing a (base) toolchain compiler like GCC.
```
Comment (akesandgren) [inline docs/System_toolchain.rst]: Shouldn't that be "a (base) toolchain compiler like GCCcore"?

File: docs/System_toolchain.rst [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+
+.. _system_toolchain_SYSTEM_constant:
+
+``SYSTEM`` constant
+-------------------
+
+For convenience, a ``SYSTEM`` constant is available to easily specify the use of the ``system`` toolchain
+(and to make it stand out against using a regular toolchain):
+
```
Comment (akesandgren) [inline docs/System_toolchain.rst]: It should be mentioned that the constant is the preferred way to do this.

File: docs/System_toolchain.rst [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+.. _system_toolchain_motivation_deprecating_dummy_name:
+
+``dummy`` toolchain name
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``dummy`` name may be rather confusing, especially if you are new to EasyBuild.
+Just based on the name, it is unclear how the ``dummy`` toolchain is different from a regular toolchain.
+
+Hopefully, the name of the ``system`` toolchain does speak more for itself: when using it, you are relying on the
+compiler & lilbraries provided by the operating system (as opposed to compilers & libraries that are a part of a regular toolchain).
```
Comment (akesandgren) [inline docs/System_toolchain.rst]: "lilbraries" -> "libraries"

File: docs/Manipulating_dependencies.rst (hunk 1 above)
Comment (boegel) [reply docs/Manipulating_dependencies.rst]: Sure should.

File: docs/Python-2-3-compatibility.rst (hunk 2 above)
Comment (boegel) [reply docs/Python-2-3-compatibility.rst]: It's a bit like magic though... Isn't it?

File: docs/System_toolchain.rst (hunk 3 above)
Comment (boegel) [reply docs/System_toolchain.rst]: `has been` works for me (`have` is plural, no?)

File: docs/System_toolchain.rst (hunk 4 above)
Comment (boegel) [reply docs/System_toolchain.rst]: gotcha, thx

File: docs/System_toolchain.rst (hunk 6 above)
Comment (boegel) [reply docs/System_toolchain.rst]: good point, consider it done

File: docs/System_toolchain.rst (hunk 7 above)
Comment (boegel) [reply docs/System_toolchain.rst]: Hmm, why didn't the spell checker that Travis runs catch this... :-/
Thx!

File: docs/Python-2-3-compatibility.rst [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+Controlling which ``python`` command EasyBuild will use via ``$EB_PYTHON``
+--------------------------------------------------------------------------
+
+The ``eb`` command will consider different ``python`` commands, and check the Python version corresponding with
+the command before selecting one. The commands considered are (in order):
+
+* ``python2``
+* ``python3``
+* ``python``
+
```
[the remark this answers is not in the document]
Comment (boegel) [reply docs/Python-2-3-compatibility.rst]: I made a mental note to fix the change in the docs too, and then forgot about it, ugh. Thanks for catching this!

File: docs/Easyconfig-files-local-variables.rst [hunk 9]
```diff
@@ -100,7 +101,7 @@ To get rid of this warning, you can either:
 
 * configure EasyBuild to only log the warning (not print it), via ``--local-var-naming-check=log``
   (see also :ref:`easyconfig_files_local_variables_check_mode`; note that silencing
-  the printed wanrnign is **not recommended**, see the motivation in :ref:`motivation above easyconfig_files_local_variables`
+  the printed wanrnign is **not recommended**, see the motivation in :ref:`motivation above easyconfig_files_local_variables_motivation`
```
[the remark this answers is not in the document]
Comment (boegel) [reply docs/Easyconfig-files-local-variables.rst]: Why didn't the spelling check done in the tests run by Travis trigger this... :(

Fixed in 3087135

File: docs/EasyBuild4-overview-of-changes.rst [hunk 10]
```diff
@@ -0,0 +1,257 @@
...
+of the core functionality of building and installing scientific software, including ``keyring`` for the GitHub
+integration features (see :ref:`integration_with_github`), etc.
+
+
+.. _eb4_python3_support:
+
+Support for running EasyBuild on top of Python 3
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A significant effort has been done to make the EasyBuild framework and the easyblocks included with EasyBuild
```
[the remark this answers is not in the document]
Comment (boegel) [reply docs/EasyBuild4-overview-of-changes.rst]: both fixed in 3087135

File: docs/EasyBuild4-overview-of-changes.rst [hunk 11]
```diff
@@ -0,0 +1,257 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A couple of **additional custom software-specific easyblocks** were added in EasyBuild v4.0, including:
+
+* ``iccifort``: to install ``icc`` and ``ifort`` together in a single prefix, and use that installation as a toolchain
+
+* ``numexpr``: to ensure that ``numexpr`` is linked with Intel MKL's VML when it is available
+
+* ``OpenMPI``: to improve the (default) configuration of ``OpenMPI`` based on installed OS packages (for example for
+  Infiniband support) and available dependencies
```
[the remark this answers is not in the document]
Comment (boegel) [reply docs/EasyBuild4-overview-of-changes.rst]: fixed in 3087135

File: docs/Python-2-3-compatibility.rst [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+  $ export EB_VERBOSE=1
+  $ eb --version
+  >> Considering 'python2'...
+  >> No 'python2' found in $PATH, skipping...
+  >> Considering 'python3'...
+  >> 'python3' version: 3.6.9, which matches Python 3 version requirement (>= 3.5)
+  >> Selected Python command: python3 (/usr/local/bin/python3)
+  >> python3 -m easybuild.main --version
+  This is EasyBuild 4.0.0 (framework: 4.0.0, easyblocks: 4.0.0) on host example.
```
[the remark this answers is not in the document]
Comment (boegel) [reply docs/Python-2-3-compatibility.rst]: Yeah, you're right, fixed in fixed in 3087135

File: docs/Python-2-3-compatibility.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (boegel) [reply docs/Python-2-3-compatibility.rst]: fixed in 3087135

File: docs/Python-2-3-compatibility.rst [hunk 13]
```diff
@@ -0,0 +1,244 @@
...
+
+* functions from the Python standard library which should be imported from different locations in Python 2 and 3
+* wrappers for functionality in the Python standard library which behaves differently in Python 2 and 3
+
+.. _py2vs3_ascii_letters:
+
+``ascii_letters``
+-----------------
+
+* Python 2: corresponds with ``string.letters``
```
[the remark this answers is not in the document]
Comment (boegel) [reply docs/Python-2-3-compatibility.rst]: done in 3087135

**Comment by migueldiascosta** [comment]
Going in, thanks @boegel!

**Pull request merged**